pkg/kubelet/kubelet.go

```diff
@@ -1715,6 +1727,42 @@ func (kl *Kubelet) syncLoop(updates <-chan PodUpdate, handler SyncHandler) {
 	}
 }
 
+// syncStatus syncs pods statuses with the apiserver.
+func (kl *Kubelet) syncStatus() {
+	glog.V(3).Infof("Syncing pods status")
+
+	// Create the list of pods.
+	podFullNames := make(map[types.UID]string)
+	func() {
+		kl.podLock.Lock()
+		defer kl.podLock.Unlock()
+		for _, pod := range kl.pods {
+			source := pod.Annotations[ConfigSourceAnnotationKey]
```

Comment:
My PR (#5401) has just been merged which enables sync status of static pods back to the apiserver. You should not need to check the source here anymore.

One caveat here is that the mirror pod at the api server may not have been created yet at the time of syncing. I think the small race window is fine for now. If we want to avoid this completely, we could look at both mirroPods and pods from kl.GetPods() to ensure the mirror pod has been created. Nonetheless, if the mirror pod gets deleted before the sync, we would still have the race. We'll just have to tolerate that.